.github/workflows/rust.yml

```diff
@@ -97,3 +97,31 @@ jobs:
       run: cargo +nightly-2025-01-03 miri setup
     - name: Run tests inside Miri
       run: cargo +nightly-2025-01-03 miri test
+
+  test-sanitizers:
+      runs-on: ubuntu-24.04
+      strategy:
+        fail-fast: false
+        matrix:
+          # TODO: Memory sanitizer is temporarity disabled because of https://github.com/cberner/fuser/pull/390
+          sanitizer: [address, thread, safestack]
+
+      env:
+        LSAN_OPTIONS: fast_unwind_on_malloc=0:max_leak_stack_depth=50:suppressions=../../../../lsan_suppressions.txt:print_suppressions=1
+        ASAN_OPTIONS: detect_leaks=1:detect_stack_use_after_return=1:check_initialization_order=1:strict_init_order=1
+      steps:
+        - uses: actions/checkout@ee0669bd1cc54295c223e0bb666b733df41de1c5 # v2
+        - uses: ./.github/actions/setup-environment
+        - name: Install Nightly Rust toolchain
+          uses: actions-rs/toolchain@v1
+          with:
+            profile: minimal
+            toolchain: nightly-2025-01-03
```

Comment:
It should use a current version of nightly as if right now (such as `nightly-2025-09-08`). I agree it should be pinned, but I don't think it has to match the MSRV actually used for builds, just for checks a new version is fine